dockstore-webservice/src/main/java/io/dockstore/webservice/core/Event.java

```diff
@@ -0,0 +1,168 @@
+    @ApiModelProperty(value = "Organisation that the event is acting on.", position = 2)
+    private Organisation organisation;
+
+    @ManyToOne
+    @JoinColumn(name = "toolId", referencedColumnName = "id")
+    @ApiModelProperty(value = "Tool that the event is acting on.", position = 3)
+    private Tool tool;
+
+    @ManyToOne
+    @JoinColumn(name = "workflowId", referencedColumnName = "id")
+    @ApiModelProperty(value = "Workflow that the event is acting on.", position = 4)
+    private Workflow workflow;
+
+    @ManyToOne
+    @JoinColumn(name = "initiatorUserId", referencedColumnName = "id")
+    @ApiModelProperty(value = "User initiating the event.", position = 5)
+    private User initiatorUser;
+
+    @Column
+    @Enumerated(EnumType.STRING)
+    @ApiModelProperty(value = "The event type.", required = true, position = 6)
+    private EventType type;
+
+    // database timestamps
+    @Column(updatable = false)
+    @CreationTimestamp
+    private Timestamp dbCreateDate;
+
+    @Column()
+    @UpdateTimestamp
+    private Timestamp dbUpdateDate;
+
+    public Event(User user, Organisation organisation, Workflow workflow, Tool tool, User initiatorUser, EventType type) {
+        this.user = user;
+        this.organisation = organisation;
+        this.workflow = workflow;
+        this.tool = tool;
+        this.initiatorUser = initiatorUser;
+        this.type = type;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    public Organisation getOrganisation() {
+        return organisation;
+    }
+
+    public void setOrganisation(Organisation organisation) {
+        this.organisation = organisation;
+    }
+
+    public Tool getTool() {
+        return tool;
+    }
+
+    public void setTool(Tool tool) {
+        this.tool = tool;
+    }
+
+    public Workflow getWorkflow() {
+        return workflow;
+    }
+
+    public void setWorkflow(Workflow workflow) {
+        this.workflow = workflow;
+    }
+
+    public User getInitiatorUser() {
+        return initiatorUser;
+    }
+
+    public void setInitiatorUser(User initiatorUser) {
+        this.initiatorUser = initiatorUser;
+    }
+
+    public Timestamp getDbCreateDate() {
+        return dbCreateDate;
+    }
+
+    public void setDbCreateDate(Timestamp dbCreateDate) {
+        this.dbCreateDate = dbCreateDate;
+    }
+
+    public Timestamp getDbUpdateDate() {
+        return dbUpdateDate;
+    }
+
+    public void setDbUpdateDate(Timestamp dbUpdateDate) {
+        this.dbUpdateDate = dbUpdateDate;
+    }
+
+    public EventType getType() {
+        return type;
+    }
+
+    public void setType(EventType type) {
+        this.type = type;
+    }
+
+    public enum EventType {
+        CREATE_ORG,
+        DELETE_ORG,
+        MODIFY_ORG,
+        APPROVE_ORG,
```

Comment:
Looking at the list of events, I wonder if there shouldn't there be an event to DISAPPROVE_ORG? If you do that, that has implications -- it means there would be state of disapproval. May be too much for phase 1, but just throwing it out there, as I could see the following use cases:

1. A user applies for org approval; the curator denies it, but then the user provides some proof that convinces the curator to approve.
2. An approved org's page has a bunch of links to evil websites (maybe they were hacked). We would to shut down the org page. Org removes links, and they get approved again.

We don't necessarily need a distinct state and event for this, but throwing it out as food for thought.

Reply:
These events are definitely preliminary and can be changed. For the DISAPPROVE_ORG, there isn't really an event as of yet that occurs. Disapproving an organisation is not yet an action. An organisation is simply not approved until it is approved. The concept of disapproving is probably useful and I think warrants some discussion, perhaps in the spec.